lib/new_sets.bzl

```diff
@@ -0,0 +1,219 @@
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Skylib module containing common hash-set algorithms.
+
+  An empty set can be created using: `sets.make()`, or it can be created with some starting values
+  if you pass it an enum: `sets.make([1, 2, 3])`. This returns a struct containing all of the values
+  as keys in a dictionary - this means that all passed in values must be hashable.  The values in
+  the set can be retrieved using `sets.to_list(my_set)`.
+
+  These functions are focused primarily t
+
+"""
+
+load(":dicts.bzl", "dicts")
+
+
+def _make(elements=None):
+  """Creates a new set.
+
+  All elements must be hashable.
+
+  Args:
+    elements: Optional enum to construct the set out of.
```

Comment:
enum? Do you been sequence?

Reply:
I meant whichever you think is more correct 😄